dev/DynamicDependency/DataStore.cpp

```diff
@@ -0,0 +1,169 @@
+
+    auto filename{ path / (packageDependency.Id() + DataStore::fileExtension) };
+
+    wil::unique_hfile file{ ::CreateFileW(filename.c_str(), GENERIC_WRITE, FILE_SHARE_DELETE, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr) };
+    if (!file)
+    {
+        THROW_LAST_ERROR_MSG("%ls", filename.c_str());
+    }
+
+    DWORD bytesWritten{};
+    THROW_IF_WIN32_BOOL_FALSE_MSG(::WriteFile(file.get(), json.c_str(), static_cast<DWORD>(json.length()), &bytesWritten, nullptr), "%ls", filename.c_str());
+}
+
+void MddCore::DataStore::Delete(PCWSTR packageDependencyId)
+{
+    std::filesystem::path relativeFilename{ L"DynamicDependency" };
+    relativeFilename /= std::wstring(packageDependencyId) + DataStore::fileExtension;
+
+    auto filename{ GetDataStorePathForUser() };
+    filename /= relativeFilename;
+    if (!DeleteFileIfExists(filename.c_str()))
+    {
+        filename = GetDataStorePathForSystem();
+        filename /= relativeFilename;
+        DeleteFileIfExists(filename.c_str());
+    }
+}
+
+bool MddCore::DataStore::DeleteFileIfExists(PCWSTR filename)
+{
+    if (!::DeleteFileW(filename))
+    {
+        auto const lastError{ GetLastError() };
+        if ((lastError == ERROR_FILE_NOT_FOUND) || (lastError == ERROR_PATH_NOT_FOUND))
+        {
+            return false;
+        }
+        THROW_HR_MSG(HRESULT_FROM_WIN32(lastError), "Error %d deleting file %ls", lastError, filename);
+    }
+    return true;
+}
+
+HANDLE MddCore::DataStore::OpenFileIfExists(
+    PCWSTR filename)
+{
+    wil::unique_hfile file{ ::CreateFileW(filename, GENERIC_READ, FILE_SHARE_READ, nullptr, OPEN_EXISTING, FILE_FLAG_SEQUENTIAL_SCAN, nullptr) };
+    if (!file)
+    {
+        const auto lastError{ GetLastError() };
+        if ((lastError == ERROR_FILE_NOT_FOUND) || (lastError == ERROR_PATH_NOT_FOUND))
+        {
+            return nullptr;
+        }
+        THROW_WIN32_MSG(lastError, "Error %d opening file %ls", lastError, filename);
+    }
+    return file.release();
+}
+
+std::filesystem::path MddCore::DataStore::GetDataStorePath(const MddCreatePackageDependencyOptions options)
+{
+    if (WI_IsFlagSet(options, MddCreatePackageDependencyOptions::ScopeIsSystem))
+    {
+        return GetDataStorePathForSystem();
+    }
+    else
+    {
+        return GetDataStorePathForUser();
+    }
+}
+
+std::filesystem::path MddCore::DataStore::GetDataStorePathForSystem()
+{
+    wil::unique_cotaskmem_ptr<WCHAR[]> folderPath;
+    THROW_IF_FAILED(SHGetKnownFolderPath(FOLDERID_ProgramData, 0, nullptr, wil::out_param(folderPath)));
+
+    std::filesystem::path path{ folderPath.get() };
+    path /= L"Microsoft";
+    return path;
+}
+
+std::filesystem::path MddCore::DataStore::GetDataStorePathForUser()
+{
+    //SEEME wil::com_ptr<IUnknown> dataStore_iunknown{ wil::CoCreateInstance<DynamicDependencyDataStore, IDynamicDependencyDataStore>(CLSCTX_LOCAL_SERVER) };
+    //SEEME auto dataStore{ dataStore_iunknown.query<IDynamicDependencyDataStore>() };
```

Comment:
[External]
What is this?

Reply:
Doh! Leftover dead code from earlier debugging. Removing.

Thanks!